pulsar-client-kafka-compat/pulsar-client-kafka/src/main/java/org/apache/kafka/clients/producer/PulsarKafkaProducer.java

```diff
@@ -76,39 +78,39 @@ public PulsarKafkaProducer(Map<String, Object> configs) {
         this(configs, null, null);
     }
 
-    public PulsarKafkaProducer(Map<String, Object> configs, Serializer<K> keySerializer,
-            Serializer<V> valueSerializer) {
+    public PulsarKafkaProducer(Map<String, Object> configs, Schema<K> keySerializer,
+            Schema<V> valueSerializer) {
         this(configs, new Properties(), keySerializer, valueSerializer);
     }
 
     public PulsarKafkaProducer(Properties properties) {
         this(properties, null, null);
     }
 
-    public PulsarKafkaProducer(Properties properties, Serializer<K> keySerializer, Serializer<V> valueSerializer) {
+    public PulsarKafkaProducer(Properties properties, Schema<K> keySerializer, Schema<V> valueSerializer) {
         this(new HashMap<>(), properties, keySerializer, valueSerializer);
     }
 
     @SuppressWarnings({ "unchecked", "deprecation" })
-    private PulsarKafkaProducer(Map<String, Object> conf, Properties properties, Serializer<K> keySerializer,
-            Serializer<V> valueSerializer) {
+    private PulsarKafkaProducer(Map<String, Object> conf, Properties properties, Schema<K> keySerializer,
+            Schema<V> valueSerializer) {
         properties.forEach((k, v) -> conf.put((String) k, v));
 
         ProducerConfig producerConfig = new ProducerConfig(conf);
 
         if (keySerializer == null) {
-            this.keySerializer = producerConfig.getConfiguredInstance(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG,
-                    Serializer.class);
-            this.keySerializer.configure(producerConfig.originals(), true);
+            this.keySerializer = new PulsarKafkaSchema<>();
+            ((PulsarKafkaSchema<K>) this.keySerializer).initSerialize(
```

Comment:
The cast seems awkward to me. Why not initialize the Serializer first and construct the schema with the serializer?

```
Serializer<T> kafkaSerializer = producerConfig.getConfiguredInstance(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, Serializer.class);
kafkaSerializer.configure(producerConfig.originals(), true);

this.keySerializer = new PulsarKafkaSchema<>(kafkaSerializer, null);
```